packages/charts/react-charts/library/src/components/CommonComponents/Annotations/ChartAnnotationLayer.tsx

```diff
@@ -539,17 +584,36 @@ export const ChartAnnotationLayer: React.FC<ChartAnnotationLayerProps> = React.m
     const baseTopLeftX = resolved.point.x + offsetX;
     const baseTopLeftY = resolved.point.y + offsetY;
 
-    const usePlotBounds = layout?.clipToBounds !== false;
-    const viewportX = usePlotBounds ? context.plotRect.x : 0;
-    const viewportY = usePlotBounds ? context.plotRect.y : 0;
-    const viewportWidth = usePlotBounds ? context.plotRect.width : context.svgRect.width ?? 0;
-    const viewportHeight = usePlotBounds ? context.plotRect.height : context.svgRect.height ?? 0;
+    const usesViewportSpace = annotation.coordinates?.type === 'relative' && layout?.clipToBounds === false;
+    const clampRect = usesViewportSpace
+      ? {
+          x: 0,
+          y: 0,
+          width:
+            typeof context.svgRect.width === 'number' && Number.isFinite(context.svgRect.width)
+              ? context.svgRect.width
+              : 0,
+          height:
+            typeof context.svgRect.height === 'number' && Number.isFinite(context.svgRect.height)
+              ? context.svgRect.height
+              : 0,
+        }
+      : layout?.clipToBounds !== false
+      ? context.plotRect
+      : undefined;
+
+    const clampX = clampRect && typeof clampRect.x === 'number' && Number.isFinite(clampRect.x) ? clampRect.x : 0;
```

Comment:
clampX clampY both values are very similar and repeating part of the code. See how we can reduce common code.